utility_analysis_new/contribution_bounders.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ContributionBounder for utility analysis."""
+import collections
+
+from pipeline_dp import contribution_bounders
+
+
+class SamplingCrossAndPerPartitionContributionBounder(
+        contribution_bounders.ContributionBounder):
+    """'Bounds' the contribution by privacy_id per and cross partitions.
+
+  Because this is for Utility Analysis, it doesn't actually ensure that
+  contribution bounds are enforced. Instead, it keeps track of probabilities
+  different data points are kept under per-partition and cross-partition
+  contribution bounding.
+
+  Only works for count at the moment.
+  """
+
+    def bound_contributions(self, col, params, backend, report_generator,
+                            aggregate_fn):
+        """See docstrings for this class and the base class."""
+        col = backend.map_tuple(
+            col, lambda pid, pk, v: (pid, (pk, v)),
+            "Rekey to ((privacy_id), (partition_key, value))")
+        col = backend.group_by_key(
+            col, "Group by key to get (privacy_id, [(partition_key, value)])")
+        report_generator.add_stage(
+            f"Converting ((privacy_id), (partition_key, value)) to"
```

Comment:
Explain computations reports are for users. So it should contain some high-level information, understandable for users. This looks pretty low-level, please drop it.